### PR TITLE
Rename `ScoreRank` to `Grade`

### PR DESCRIPTION
**Pull request by peppy**
Because I spend minutes trying to find this each time.

I think it's unanimously referrred to as a grade these days. The only places I can recall where it's called "rank" is at song select / beatmap listing:

Safari 2023-09-19 at 06 43 10

Jump Desktop 2023-09-19 at 06 43 29

And maybe occasionally used in conversation as "S rank". But the more common usage of "rank" in the community is "global rank". Maybe we can use this opportunity to fix this naming conflict going foward?

@ppy/team-client RFC to make sure everyone's on board with this. if we make this change, it will not match database usage (which will remain as `rank`) for the time being. It will also be a breaking change for rulesets and external tools.

### CODE REVIEW

**Review by smoogipoo — COMMENTED** [review]
I'm fine with this. Also considering `ScoreGrade` but I have no hard feelings.

**Comment by peppy** [comment]
> I'm fine with this. Also considering `ScoreGrade` but I have no hard feelings.

On the contrary, if we're leaving as `ScoreGrade` I'd consider just dropping this change. Not sure why, the prefix bugs me.

File: osu.Game/Graphics/OsuColour.cs [hunk 1]
```diff
@@ -40,27 +40,27 @@ public class OsuColour
         }, (float)Math.Round(starDifficulty, 2, MidpointRounding.AwayFromZero));
 
         /// <summary>
-        /// Retrieves the colour for a <see cref="ScoreRank"/>.
+        /// Retrieves the colour for a <see cref="Grade"/>.
         /// </summary>
-        public static Color4 ForRank(ScoreRank rank)
+        public static Color4 ForRank(Grade grade)
```
[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game/Graphics/OsuColour.cs:45]: Probably a few edge cases I missed.

**Comment by bdach** [comment]
In principle and in isolation I don't hate this change, but the interactions with everything else give me pause.

Copy can be adjusted, that's no big deal, but the database and API structures not changing in kind causes a situation where now both terms are in use for this one thing. This becomes even more painful with stuff like JSON serialization where if we aren't careful (and historically we haven't always 100% been), we'll get regressions out of this because of attempting to read data from the wrong place.

For another data point the wiki already uses "grade". Which doesn't make any of this any easier.

Overall, kinda not sure the change is worth it given how much time we'll have to put into migrating and making sure we haven't accidentally broken stuff... that said again, even if I'm opposed, then at most weakly.

**Comment by bdach** [comment]
@peppy I think you polled twitter for this, how'd that go? Can't check myself for obvious reasons.

**Comment by peppy** [comment]
Was a bit of a mixed bag. Most people agreed with "grade" but there was a large group which was like "why change"?

Safari 2023-09-25 at 08 38 39

I don't think this is actionable for now so will close this PR for the time being.

**Pull request closed without merging**